torch/csrc/jit/tensorexpr/loopnest.cpp

```diff
@@ -2030,6 +2025,68 @@ class StoreFinder : public IRVisitor {
   const Store* store_;
 };
 
+class BufReplacer : public IRMutator {
+ public:
+  BufReplacer(
+      const Buf* old_buf,
+      const std::vector<const Expr*>& old_indices,
+      const Buf* new_buf,
+      const std::vector<const Expr*>& new_indices)
+      : old_buf_(old_buf),
+        old_indices_(old_indices),
+        new_buf_(new_buf),
+        new_indices_(new_indices) {}
+
+  const Expr* mutate(const Load* v) override {
+    if (v->buf() != old_buf_) {
+      return IRMutator::mutate(v);
+    }
+
+    TORCH_INTERNAL_ASSERT(old_indices_.size() == v->indices().size());
+
+    bool equal_indices = true;
+    for (size_t i = 0; i < v->indices().size(); ++i) {
+      if (!exprEquals(v->indices()[i], old_indices_[i])) {
+        equal_indices = false;
+        break;
+      }
+    }
+    if (!equal_indices) {
+      return IRMutator::mutate(v);
+    }
+
+    return new Load(new_buf_, new_indices_, v->mask());
```

Comment:
Is it worth recursing on `v->mask()` ?, and do we not need to check that they're equal?

Reply:
Hmm, perhaps. It would be very weird if it mattered - the mask would have to include a Load to the same Buf as the Load it belongs to. I can't think of any legitimate reason that would occur.

Actually masks probably shouldn't be dynamic, that way lies madness. It's not a big deal to change it here though I guess.